sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/utils-service/_tests_/unit/application-event-code-during-enrolment-and-completed-utils.service.applicationEventCodeDuringCompleted.spec.ts

```diff
@@ -0,0 +1,219 @@
+import {
+  COEStatus,
+  DisbursementScheduleStatus,
+  RestrictionActionType,
+} from "@sims/sims-db";
+import {
+  ApplicationEventCode,
+  DisbursementScheduleForApplicationEventCodeDuringCompleted,
+} from "../../../models/ier12-integration.model";
+import { DisbursementValueService } from "@sims/integrations/services";
+import { ApplicationEventCodeDuringEnrolmentAndCompletedUtilsService } from "../..";
+import { DISBURSEMENT_FILE_GENERATION_ANTICIPATION_DAYS } from "@sims/services/constants";
+import { DATE_ONLY_ISO_FORMAT, addDays, formatDate } from "@sims/utilities";
+import { FULL_TIME_DISBURSEMENT_FEEDBACK_ERRORS } from "@sims/integrations/services/disbursement-schedule/disbursement-schedule.models";
+import { createMock } from "@golevelup/ts-jest";
+
+describe("applicationEventCodeDuringEnrolmentAndCompletedUtilsService-applicationEventCodeDuringCompleted", () => {
+  let applicationEventCodeDuringEnrolmentAndCompletedUtilsService: ApplicationEventCodeDuringEnrolmentAndCompletedUtilsService;
+  let disbursementValueService: DisbursementValueService;
+  let payload: DisbursementScheduleForApplicationEventCodeDuringCompleted;
+
+  beforeAll(() => {
+    disbursementValueService = createMock<DisbursementValueService>();
+    applicationEventCodeDuringEnrolmentAndCompletedUtilsService =
+      new ApplicationEventCodeDuringEnrolmentAndCompletedUtilsService(
+        disbursementValueService,
+      );
+    payload = {
+      id: 9999,
+      coeStatus: COEStatus.completed,
+      disbursementDate: formatDate(new Date(), DATE_ONLY_ISO_FORMAT),
+      disbursementScheduleStatus: DisbursementScheduleStatus.Pending,
+    };
+  });
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it(`Should return ${ApplicationEventCode.DISC} when the disbursement schedule status is ${DisbursementScheduleStatus.Cancelled}.`, async () => {
+    // Arrange and act.
```

Comment:
As a suggestion, as much as possible I would like to keep the Arrange/Act/Assert section isolated, as below.
```ts
// Arrange
const currentDisbursementSchedule = {
  ...payload,
  disbursementScheduleStatus: DisbursementScheduleStatus.Cancelled,
};

// Act
const applicationEventCode =
  await applicationEventCodeDuringEnrolmentAndCompletedUtilsService.applicationEventCodeDuringCompleted(
    currentDisbursementSchedule,
  );

// Assert.
expect(applicationEventCode).toBe(ApplicationEventCode.DISC);
```